### ISSUE TITLE
Minor Error Handling Improvement: Indicate URL in non-200 status error message

### ISSUE THREAD

**Issue by davewichers** [dependency-check/DependencyCheck#1182]
Line ~273 of utils/Downloader.java is like this:
  throw new DownloadFailedException(format("%s request returned a non-200 status code", httpMethod));

Problem with this is that is simply prints out an error message like this:

[ERROR] IO Exception: HEAD request returned a non-200 status code

without telling you which URL it was trying to access, which makes it hard to debug. Can you include the URL being accessed in this error message? And any other similar error messages generated by DependencyCheck?

Thx

**Comment by davewichers** [comment]
Maybe something as simple as:
throw new DownloadFailedException(format("%s request returned a non-200 status code for URL: " + url, httpMethod));

This worked for me. But I don't know if this is the right way to do this in DependencyCheck.

**Comment by davewichers** [comment]
I also noticed a similar error, but don't know how to fix it. In this case, at the very end of a DC run, I'm getting the error:

[ERROR] java.security.cert.CertificateException: No name matching api.nodesecurity.io found

But it's giving me no clue as to where it's trying to go (I'm assuming: https://api.nodesecurity.io/check) and why (i.e., what switch in DC do I have to change or turn OFF to make this error go away in my environment).

I have already turned this off: nodeAnalyzerEnabled. I suspect I need to also turn this off: nspAnalyzerEnabled. But it would be very helpful if the error message explained this to the user.
